samza-test/src/main/java/org/apache/samza/test/framework/TestRunner.java

```diff
@@ -229,8 +241,8 @@ public TestRunner addOutputStream(InMemoryOutputDescriptor streamDescriptor, int
     factory
         .getAdmin(streamDescriptor.getSystemName(), config)
         .createStream(spec);
-    addConfigs(streamDescriptor.toConfig());
-    addConfigs(streamDescriptor.getSystemDescriptor().toConfig());
+    putIfAbsentConfig(streamDescriptor.toConfig());
```

Comment:
Can we add to the Javadoc of this method to explain why we are calling putIfAbsent here? I assume the reason is to allow user supplied config to supersede descriptor generated config? Can we make it clear?